index.d.ts

```diff
@@ -1,17 +1,20 @@
-import { Dispatch, Store, Unsubscribe } from 'redux';
+import { Dispatch, Store, Unsubscribe, AnyAction } from 'redux';
 
 export function shallowEquals(a: any, b: any): boolean;
 
-interface Options {
+type Options = {
   skipInitialCall?: boolean;
   equals?: typeof shallowEquals;
 }
 
-type Observer<S = any, MS = any> = unknown;
```

Comment:
The  `observer()` "constructor" function (defined below) returns a function that takes `(state, dispatch, options)`. The function itself was not intended to be called by client code, but I think a more descriptive type alias would be helpful here. Moreover, allowing the user to call it would not cause any issues.

I've elided `options` from the signature as I personally find it a bit confusing to leave them in. Those options would be considered `globals` (those passed in to `observe()`). Anyway, `options` could be accommodated, too, if need be.